### DISCUSSION TITLE
Send request to an URL with JSON data with flag color from race control message

### DISCUSSION THREAD

**Discussion by stefanverkerknl** [f1multiviewer/issue-tracker#60] [Feature requests] [answered]
This idea is a wild one, but I think a lot of users with smart lights (Philips Hue, Nanoleaf, LIFX) would love this one and for me personally this would be a dream come true (well, sort of).  

I use Philips Hue in my home for my lights and as a webdesigner I've made a webtool to turn on multiple lights when I push a button in the web app. I use the Philips Hue API for this purpose. 

In my room where I watch F1TV, I have multiple lights which I can trigger with this app. I can give the lights colors, adjust the brightness etc. I would really like it to have an option in MultiViewer that when I've got the Live Timing window open and there's a certain flag (for example yellow, green, red), MultiViewer sends a request to a given URL. That way I can change the color of my lights to for example yellow when there's a yellow flag to create an even better experience. And in my case it's Philips Hue, but I can imagine that people with wall tiles (Nanoleaf, LIFX) would love to create a yellow or red background when there's a yellow or red flag.  

For example: in the Live Timing window there's a race control message which says: 'yellow flag at sector 1'. When that message appears, MultiViewer sends a request from the MultiViewer app to a by user given URL (given in a settings page of the app) with the following JSON data (for example with a POST request or GET request):

`{
	"flag_color": "yellow"
}`

With a selfmade script on my own server, I can handle that request (for example in PHP), get the color of the flag from the JSON data and turn my lights on and switch them to yellow for a few seconds with the Philips Hue API. 

So this means that MultiViewer doesn't have to do any extra stuff, it just has to send a request to a by user given URL with the JSON data with the color of the flag. In my case I would like to use my own script to work with the Philips Hue API, but I can think of people who rather would like to use If This Then That for it for example. 

And by the way, this could be extended to other colors of course. For example, when somebody sets the fastest lap of the race, send the color 'purple'. 

I hope this is something you can put in the app in the future. Thanks!

![flags](https://user-images.githubusercontent.com/18143837/174473763-c59d1636-a974-48c6-8a5f-2d9c972c1725.jpg)

**Comment by JJWatMyself** [comment]
We didn't give credit over here yet, but there was so much interest on the Discord to set color on lights to match the flag color.  Encourage all with colored lights to stop by here and add their vote for visibility.

My area of interest is Philips Hue.

SIMPLE: Hue Sync for Windows/Mac and flag color updated on live timing window background, where live timing is on second display for Hue Sync to grab dominant screen color.
BETTER: configurable URLs for flag colors and some sample URLs for various platforms, e.g. Hue Sync

**Comment by bartlangelaan** [comment]
I guess this could be useful for more then only flag color. If it were possible to provide an API url in the settings, then more data could be sent there in the future. 

I can imagine that you want to flash your light when the fastest lap time changes. Which whould be possible if data was sent like:
```
{
  "type": "RACE_CONTROL_MESSAGE", 
  "message": "yellow flag at sector 1"
}
```

```
{
  "type": "FASTEST_LAP", 
  "driver": "[your favorite driver]",
  "time": 100.00
}
```

By just using one endpoint and sending multiple update types, it is possible to go wild on automations. If the API would only receive the color, it is harder to create your own customized experience.

Also, in Home Assistant you can easily receive webhook data and create automations based on that. This makes it possible to customize all of this without having to program your own server. Just checking what kind of data was sent, and deciding logic based on that, is supported out of the box.

**Comment by f1multiviewer** [comment, marked as answer]
While there is no webhooks yet, a simple API has been added (which allows access to timing state, you can see more in #technical channel)

**Reply by f1multiviewer** [reply]
I'm gonna consider this done at least for now. Improvements are coming, but I think the core of this FR is covered with the current functionality :)
